src/tap/grpc/server.rs

```diff
@@ -125,10 +142,24 @@ where
             }
         };
 
+        let extract = match req.extract.and_then(|ex| ExtractKind::try_from(ex).ok()) {
+            Some(ex) => ex,
+            None => {
+                // If there's no extract field, the request may have been sent
+                // by an older version of the Linkerd control plane. If this is
+                // the case, rather than failing the tap, just do the only
+                // behavior that older control planes know about --- extract
+                // HTTP data without headers.
+                let default = ExtractKind::Http { headers: false };
+                warn!(?default, "tap request with no extract field; using default");
```

Comment:
Ehhhh, I feel uneasy about emitting warnings to users with debug output like this in this case... It seems like something we should either panic on, because we expect never to encounter it, or to keep it at debug so users don't see it and get concerned, when they can't control anything that interacts with this.

Reply:
Furthermore, are we really expecting that we'll upgrade all tap request to set this value? That was not what I expected, but we can do that if it's necessary.